packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1788,6 +1742,27 @@ export class StripeHelper {
         new Error(`Unexpected line item: ${invoice.lines.data[0].id}`)
       );
     }
+
+    const { uid, email } = await getUidAndEmailByStripeCustomerId(
+      invoice.customer
+    );
+
+    if (!email || !uid) {
+      throw error.internalValidationError(
+        'extractInvoiceDetailsForEmail',
+        { customerId: invoice.customer },
+        'Customer missing email or uid.'
+      );
+    }
+
+    const customer = await this.customer({
+      uid,
+      email,
+    });
```

Comment:
Since I now have the uid and email from my new helper, I can call `this.customer` from the Stripe Helper which at least has a chance to pull from the Redis cache before fetching a customer from Stripe (unlike `expandResouce`, which was used before).